test/widgets/test_sep.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2015 Tycho Andersen
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# Widget specific tests
+
+import pytest
+
+import libqtile.bar
+import libqtile.config
+import libqtile.confreader
+import libqtile.layout
+from libqtile import widget
+
+
+class MinimalConf(libqtile.confreader.Config):
+    auto_fullscreen = False
+    keys = []
+    mouse = []
+    groups = [libqtile.config.Group("a")]
+    layouts = [libqtile.layout.stack.Stack(num_stacks=1)]
+    floating_layout = libqtile.resources.default_config.floating_layout
+    screens = []
+
+
+sep = widget.Sep()
+
+parameters = [
+    (libqtile.config.Screen(top=libqtile.bar.Bar([sep], 10)), "top", "width"),
+    (libqtile.config.Screen(left=libqtile.bar.Bar([sep], 10)), "left", "height")
+]
+
+
+@pytest.mark.parametrize("screen,location,attribute", parameters)
+def test_orientations(manager_nospawn, screen, location, attribute):
+    config = MinimalConf
+    config.screens = [screen]
+
+    manager_nospawn.start(config)
+    bar = manager_nospawn.c.bar[location]
+
+    w = bar.info()["widgets"][0]
+    assert w[attribute] == 3
```

Comment:
I got a bit confused by the sep code. `length` seems to be set twice with different values:
https://github.com/qtile/qtile/blob/48f7c5e499fed64913732896a5fdc65fa921d2e8/libqtile/widget/sep.py#L52-L55

I would have thought the length of the widget should be `2x padding + line width`, but it's actually just `padding + line width`.

Reply:
It would not surprise me if there's a bug in the calculations here.

Reply:
OK - will take a look separately and can submit a PR if it needs fixing.